### PR TITLE
Update various docstrings, add workflow to check docstrings

### PR DESCRIPTION
**Pull request by till-m**
see #427

### CODE REVIEW

File: bayes_opt/bayesian_optimization.py [hunk 1]
```diff
@@ -233,7 +277,13 @@ def suggest(self, utility_function):
...
 
     def _prime_queue(self, init_points):
-        """Make sure there's something in the queue at the very beginning."""
+        """Ensure the queue is not empty.
+
+        Parameters
+        ----------
+        init_points: int
+            Number of parameters to prime the queue with.
+        """
```
Comment (leandrobbraga) [inline bayes_opt/bayesian_optimization.py:286]: Is this *really* what this function does? The term "prime" gives me the impression that this is preparing/driving the queue to the start point.

**Review by leandrobbraga — APPROVED** [review]

File: bayes_opt/bayesian_optimization.py [hunk 2]
```diff
@@ -167,30 +182,53 @@ def __init__(self,
...
+        constraint_value: float or None
+            Value of the constraint function at the observation, if any.
+        """
         self._space.register(params, target, constraint_value)
         self.dispatch(Events.OPTIMIZATION_STEP)
 
     def probe(self, params, lazy=True):
-        """
-        Evaluates the function on the given points. Useful to guide the optimizer.
+        """Evaluate the function on the given points.
```
Comment (leandrobbraga) [inline bayes_opt/bayesian_optimization.py:229]:
```suggestion
        """Evaluate the function at the given points.
```

**Review by leandrobbraga — CHANGES_REQUESTED** [review]

File: bayes_opt/bayesian_optimization.py (hunk 1 above)
Comment (till-m) [reply bayes_opt/bayesian_optimization.py:286]: The code is pretty short and fits here:
```python


    def _prime_queue(self, init_points):
        """Ensure the queue is not empty.

        Parameters
        ----------
        init_points: int
            Number of parameters to prime the queue with.
        """
        if self._queue.empty and self._space.empty:
            init_points = max(init_points, 1)

        for _ in range(init_points):
            self._queue.add(self._space.random_sample())
```

So what happens is a) if the queue is empty and there are no points registered to the target space it samples at least one point randomly from the space and adds it to the queue. I'm open to changing the description (or function name) to something that is more intuitive. Any suggestions?

> The term "prime" gives me the impression that this is preparing/driving the queue to the start point.

Not sure what exactly you mean by this, I think the `prime` is used here as a stand-in for `prepare`.

**Comment by bwheelz36** [comment]
Hey @till-m, sorry I don't have time to review in detail - great work on pushing this forward!

Just a query on the pydocstring - I just had a look at running it locally and it seems extremely sensitive - producing hundreds and hundreds of errors. I'm not sure if they would all have to be fixed for the job to pass? 
A lot about things I don't really care about, like missing spaces, missing full stops etc. It might be possible to configure this so it only catches things we really care about - which for me is (1) docstring missing and (2) undescribed parameters.

**Comment by till-m** [comment]
@bwheelz36, it's indeed super sensitive. I think there's two possibilities:
1) Once all the docstrings adhere to the numpy standard, there should ideally not be a lot of work in maintaining that.
2) use `--select=CODE01,CODE02,ETC` to select only specific things to check against.

I think there's an argument to be made in staying consistent with other scientific packages, lots of which use numpy's style. At the same time, I kind of agree that it seems to be overly pedantic. I would be fine with going either direction.

File: bayes_opt/bayesian_optimization.py (hunk 1 above)
Comment (leandrobbraga) [reply bayes_opt/bayesian_optimization.py:286]: I think it's fine

**Pull request merged**